src/main/java/seedu/taskman/logic/commands/Command.java

```diff
@@ -1,15 +1,27 @@
 package seedu.taskman.logic.commands;
 
+import java.util.concurrent.LinkedBlockingDeque;
+
 import seedu.taskman.commons.core.EventsCenter;
 import seedu.taskman.commons.core.Messages;
 import seedu.taskman.commons.events.ui.IncorrectCommandAttemptedEvent;
 import seedu.taskman.model.Model;
 
 /**
  * Represents a command with hidden internal logic and the ability to be executed.
+ * 
+ * Command History:
+ * LinkedBlockingDeque
+ * Front --	---- ---- ---- ---- ---- ---- ---- ---- Back
+ * offerFirst/pollFirst/pop	---	---- ---- ---- ---- offerLast/pollLast
+ * Most recently executed command -- ---- ---- ---- 10th most recently executed command
  */
 public abstract class Command {
     protected Model model;
+	private static LinkedBlockingDeque<String> inputHistory;
```

Comment:
Oh you're right about CommandParser. My miss there.
 
Should have pasted this along.. http://stackoverflow.com/a/7026563
Essentially, I'm not quite sold with using a static variable for the queue in terms of design.
We could still keep it if there's a good reason though!